src/profile.c

```diff
@@ -167,14 +168,18 @@ void Profile_Print(RedisModule_Reply *reply, void *ctx) {
         }
 
       // Print whether a warning was raised throughout command execution
+      bool warningRaised = bgScanOOM || queryOOM || timedout || reachedMaxPrefixExpansions;
       if (bgScanOOM) {
         RedisModule_ReplyKV_SimpleString(reply, "Warning", QUERY_WINDEXING_FAILURE);
       }
+      if (queryOOM) {
+        RedisModule_ReplyKV_SimpleString(reply, "Warning", QUERY_WOOM_CLUSTER);
+      }
       if (timedout) {
         RedisModule_ReplyKV_SimpleString(reply, "Warning", QueryError_Strerror(QUERY_ETIMEDOUT));
       } else if (reachedMaxPrefixExpansions) {
         RedisModule_ReplyKV_SimpleString(reply, "Warning", QUERY_WMAXPREFIXEXPANSIONS);
-      } else {
+      } else if (!warningRaised) {
```

Comment:
Either reply with only one warning (with a single if..else-if..else statement), or we start replying with an array